# Remove the tests of test_display.py that depen on cv2.SURF (unfree).
--- SimpleCV/tests/test_display.py	2014-10-03 00:53:08.970024985 +0200
+++ SimpleCV/tests/test_display.py	2014-10-03 00:55:03.237200121 +0200
@@ -1153,39 +1153,6 @@
   else:
     assert False
 
-def test_findKeypoints():
-  try:
-    import cv2
-  except:
-    pass
-    return 
-  img = Image(testimage2)
-  kp = img.findKeypoints()
-  for k in kp:
-    k.getObject()
-    k.descriptor()
-    k.quality()
-    k.octave()
-    k.flavor()
-    k.angle()
-    k.coordinates()
-    k.draw()
-    k.distanceFrom()
-    k.meanColor()
-    k.area()
-    k.perimeter()
-    k.width()
-    k.height()
-    k.radius()
-    k.crop()
-
-  kp.draw()
-  results = [img]
-  name_stem = "test_findKeypoints"
-  perform_diff(results,name_stem)    
-  
-  pass
-
 def test_movement_feature():
   current1 = Image("../sampleimages/flow_simple1.png")
   prev = Image("../sampleimages/flow_simple2.png")
@@ -1235,104 +1202,6 @@
 
   pass 
 
-def test_keypoint_extraction():
-  try:
-    import cv2
-  except:
-    pass
-    return 
-
-  img1 = Image("../sampleimages/KeypointTemplate2.png")
-  img2 = Image("../sampleimages/KeypointTemplate2.png")
-  img3 = Image("../sampleimages/KeypointTemplate2.png")
-
-  kp1 = img1.findKeypoints()
-  kp2 = img2.findKeypoints(highQuality=True)
-  kp3 = img3.findKeypoints(flavor="STAR")
-  kp1.draw()
-  kp2.draw()
-  kp3.draw()
-  #TODO: Fix FAST binding
-  #~ kp4 = img.findKeypoints(flavor="FAST",min_quality=10)
-  if( len(kp1)==190 and 
-      len(kp2)==190 and
-      len(kp3)==37
-      #~ and len(kp4)==521
-    ):
-    pass
-  else:
-    assert False
-  results = [img1,img2,img3]
-  name_stem = "test_keypoint_extraction"
-  perform_diff(results,name_stem,tolerance=3.0)    
-
-
-def test_keypoint_match():
-  try:
-    import cv2
-  except:
-    pass
-    return 
-
-  template = Image("../sampleimages/KeypointTemplate2.png")
-  match0 = Image("../sampleimages/kptest0.png")
-  match1 = Image("../sampleimages/kptest1.png")
-  match3 = Image("../sampleimages/kptest2.png")
-  match2 = Image("../sampleimages/aerospace.jpg")# should be none 
-
-  fs0 = match0.findKeypointMatch(template)#test zero
-  fs1 = match1.findKeypointMatch(template,quality=300.00,minDist=0.5,minMatch=0.2)
-  fs3 = match3.findKeypointMatch(template,quality=300.00,minDist=0.5,minMatch=0.2)
-  print "This should fail"
-  fs2 = match2.findKeypointMatch(template,quality=500.00,minDist=0.2,minMatch=0.1)
-  if( fs0 is not None and fs1 is not None and fs2 is None and  fs3 is not None):
-    fs0.draw()
-    fs1.draw()
-    fs3.draw()
-    f = fs0[0] 
-    f.drawRect()
-    f.draw()
-    f.getHomography()
-    f.getMinRect()
-    f.meanColor()
-    f.crop()
-    f.x
-    f.y
-    f.coordinates()
-  else:
-    assert False
-
-  results = [match0,match1,match2,match3]
-  name_stem = "test_find_keypoint_match"
-  perform_diff(results,name_stem)    
- 
-
-def test_draw_keypoint_matches():
-  try:
-    import cv2
-  except:
-    pass
-    return
-  template = Image("../sampleimages/KeypointTemplate2.png")
-  match0 = Image("../sampleimages/kptest0.png")
-  result = match0.drawKeypointMatches(template,thresh=500.00,minDist=0.15,width=1)
-
-  results = [result]
-  name_stem = "test_draw_keypoint_matches"
-  perform_diff(results,name_stem,tolerance=4.0)    
-  pass  
-
-def test_skeletonize():
-  img = Image(logo)
-  s = img.skeletonize()
-  s2 = img.skeletonize(10)
-
-  results = [s,s2]
-  name_stem = "test_skelotinze"
-  perform_diff(results,name_stem)    
-
-  pass
-
 def test_smartThreshold():
   img = Image("../sampleimages/RatTop.png")
   mask = Image((img.width,img.height))
@@ -1527,74 +1396,6 @@
   name_stem = "test_lineCrop"
   perform_diff(results,name_stem,tolerance=3.0)        
   pass
-    
-def test_on_edge():
-  img1 = "./../sampleimages/EdgeTest1.png"
-  img2 = "./../sampleimages/EdgeTest2.png"
-  imgA = Image(img1)
-  imgB = Image(img2)
-  imgC = Image(img2)
-  imgD = Image(img2)
-  imgE = Image(img2)
-  
-  blobs = imgA.findBlobs()
-  circs = imgB.findCircle(thresh=200)
-  corners = imgC.findCorners()
-  kp = imgD.findKeypoints()
-  lines = imgE.findLines()
-  
-  rim =  blobs.onImageEdge()
-  inside = blobs.notOnImageEdge()
-  rim.draw(color=Color.RED)
-  inside.draw(color=Color.BLUE)
-
-  rim =  circs.onImageEdge()
-  inside = circs.notOnImageEdge()
-  rim.draw(color=Color.RED)
-  inside.draw(color=Color.BLUE)
-
-  #rim =  corners.onImageEdge()
-  inside = corners.notOnImageEdge()
-  #rim.draw(color=Color.RED)
-  inside.draw(color=Color.BLUE)
-
-  #rim =  kp.onImageEdge()
-  inside = kp.notOnImageEdge()
-  #rim.draw(color=Color.RED)
-  inside.draw(color=Color.BLUE)
-
-  rim =  lines.onImageEdge()
-  inside = lines.notOnImageEdge()
-  rim.draw(color=Color.RED)
-  inside.draw(color=Color.BLUE)
-  
-  results = [imgA,imgB,imgC,imgD,imgE]
-  name_stem = "test_onEdge_Features"
-  perform_diff(results,name_stem,tolerance=7.0)        
-
-def test_feature_angles():
-  img = Image("../sampleimages/rotation2.png")
-  img2 = Image("../sampleimages/rotation.jpg")
-  img3 = Image("../sampleimages/rotation.jpg")
-  b = img.findBlobs()
-  l = img2.findLines()
-  k = img3.findKeypoints()
-
-  for bs in b:
-    tl = bs.topLeftCorner()
-    img.drawText(str(bs.angle()),tl[0],tl[1],color=Color.RED)
-  
-  for ls in l:
-    tl = ls.topLeftCorner()
-    img2.drawText(str(ls.angle()),tl[0],tl[1],color=Color.GREEN)
-
-  for ks in k:
-    tl = ks.topLeftCorner()
-    img3.drawText(str(ks.angle()),tl[0],tl[1],color=Color.BLUE)
-
-  results = [img,img2,img3]
-  name_stem = "test_feature_angles"
-  perform_diff(results,name_stem,tolerance=9.0)        
 
 def test_feature_angles_rotate():
   img = Image("../sampleimages/rotation2.png")
@@ -1707,32 +1508,3 @@
     name_stem = "test_sobel"
     s = [s]
     perform_diff(s,name_stem)
-
-def test_image_new_smooth():
-  img = Image(testimage2)
-  result = []
-  result.append(img.medianFilter())
-  result.append(img.medianFilter((3,3)))
-  result.append(img.medianFilter((5,5),grayscale=True))
-  result.append(img.bilateralFilter())
-  result.append(img.bilateralFilter(diameter=14,sigmaColor=20, sigmaSpace=34))
-  result.append(img.bilateralFilter(grayscale=True))
-  result.append(img.blur())
-  result.append(img.blur((5,5)))
-  result.append(img.blur((3,5),grayscale=True))
-  result.append(img.gaussianBlur())  
-  result.append(img.gaussianBlur((3,7), sigmaX=10 , sigmaY=12))  
-  result.append(img.gaussianBlur((7,9), sigmaX=10 , sigmaY=12, grayscale=True))  
-  name_stem = "test_image_new_smooth"
-  perform_diff(result,name_stem)
-  pass
-
-def test_track():
-    ts = []
-    bb = (195, 160, 49, 46)
-    imgs = [Image(img) for img in trackimgs]
-    ts = imgs[0].track("camshift", ts, imgs[1:], bb)
-    if ts:
-        pass
-    else:
-        assert False
